src/manifolds/ProjectiveSpace.jl

```diff
@@ -0,0 +1,392 @@
+        @invoke_maker 1 Manifold injectivity_radius(
+            M::AbstractProjectiveSpace,
+            rm::AbstractRetractionMethod,
+        )
+    end,
+)
+
+function inverse_retract!(
+    ::AbstractProjectiveSpace,
+    X,
+    p,
+    q,
+    ::Union{ProjectionInverseRetraction,PolarInverseRetraction,QRInverseRetraction},
+)
+    signz = sign(dot(p, q))
+    X .= q .* signz' .- p
+    return X
+end
+
+function Base.isapprox(::AbstractProjectiveSpace, p, q; kwargs...)
+    return isapprox(abs2(dot(p, q)), 1; kwargs...)
+end
+
+function log!(M::AbstractProjectiveSpace, X, p, q)
+    z = dot(p, q)
+    cosθ = abs(z)
+    signz = sign_from_abs(z, cosθ)
+    X .= (q .* signz' .- cosθ .* p) ./ usinc_from_cos(cosθ)
+    return project!(M, X, p, X)
+end
+
+@doc raw"""
+    manifold_dimension(M::AbstractProjectiveSpace{𝔽}) where {𝔽}
+
+Return the real dimension of the [`AbstractProjectiveSpace`](@ref) `M`, respectively i.e.
+the real dimension of the embedding minus the real dimension of the field `𝔽`.
+"""
+function manifold_dimension(M::AbstractProjectiveSpace{𝔽}) where {𝔽}
+    return manifold_dimension(get_embedding(M)) - real_dimension(𝔽)
+end
+
+"""
+    mean(
+        M::AbstractProjectiveSpace,
+        x::AbstractVector,
+        [w::AbstractWeights,]
+        method = GeodesicInterpolationWithinRadius(π/4);
+        kwargs...,
+    )
+
+Compute the Riemannian [`mean`](@ref mean(M::Manifold, args...)) of points in vector `x`
+using [`GeodesicInterpolationWithinRadius`](@ref).
+"""
+mean(::AbstractProjectiveSpace, ::Any...)
+
+function Statistics.mean!(
+    M::AbstractProjectiveSpace,
+    p,
+    x::AbstractVector,
+    w::AbstractVector;
+    kwargs...,
+)
+    return mean!(M, p, x, w, GeodesicInterpolationWithinRadius(π / 4); kwargs...)
+end
+
+function mid_point!(M::ProjectiveSpace, q, p1, p2)
+    z = dot(p1, p2)
+    cosθ = abs(z)
+    signz = sign_from_abs(z, cosθ)
+    q .= p1 .+ p2 .* signz'
+    project!(M, q, q)
+    return q
+end
+
+"""
+    normal_tvector_distribution(M::ProjectiveSpace{n,ℝ}, p, σ)
+
+Generate a distribution in the tangent space at `p` by generating a
+normal distribution in ambient space with standard deviation `σ`
+projected to the tangent space at `p`.
+"""
+function normal_tvector_distribution(M::ProjectiveSpace{n,ℝ}, p, σ) where {n}
+    d = Distributions.MvNormal(zero(p), σ)
+    return ProjectedFVectorDistribution(TangentBundleFibers(M), p, d, project!, p)
+end
+
+project!(::AbstractProjectiveSpace, q, p) = (q .= p ./ norm(p))
+
+project!(::AbstractProjectiveSpace, Y, p, X) = (Y .= X .- p .* dot(p, X))
+
+@doc raw"""
+    representation_size(M::AbstractProjectiveSpace)
+
+Return the size points on the [`AbstractProjectiveSpace`](@ref) `M` are represented as,
+i.e., the representation size of the embedding.
+"""
+@generated representation_size(::ArrayProjectiveSpace{N}) where {N} = size_to_tuple(N)
+@generated representation_size(::ProjectiveSpace{N}) where {N} = (N + 1,)
+
+function retract!(
```

Comment:
For me, this just looks like the `ProjectionRetraction`, why should it also be triggered by QU or Polar? I can't see that here.

Reply:
For `Grassmann(n,1)`, all three retractions converge (with maybe different sign conventions between `QRRetraction` and `PolarRetraction`). The `PolarRetraction` for `Grassmann` is I am quite sure also its `ProjectionRetraction` (`project` isn't implemented for `Grassmann`, but it is for `GeneralizedGrassmann`, and that's what `PolarRetraction` is doing).

Reply:
With “converge” you mean they are all the same? (Just my english maybe, I think of a limit-thingy of analysis/continuity when reading converge and would call this, they “fall together” would be quite german maybe, so maybe coincide?)

Then a small note in the docs would be nice, that they do coincide.

Reply:
Yeah, that's right, coincide is perhaps a better way of saying it.